SQL 4.4 exception when closing query

On Jet 4.4, if I query an infinite source and run

```
sqlResult.close();
```

I get lots of these in the server logs

```
Caused by: com.hazelcast.sql.impl.QueryException: Query was cancelled by the user
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:55)
	at com.hazelcast.sql.impl.QueryException.cancelledByUser(QueryException.java:91)
	at com.hazelcast.jet.sql.impl.JetSqlResultImpl.close(JetSqlResultImpl.java:78)
	at com.hazelcast.sql.impl.AbstractSqlResult.close(AbstractSqlResult.java:41)
	at com.hazelcast.sql.impl.state.QueryClientStateRegistry.close(QueryClientStateRegistry.java:178)
	at com.hazelcast.sql.impl.client.SqlCloseMessageTask.call(SqlCloseMessageTask.java:40)
	at com.hazelcast.client.impl.protocol.task.AbstractCallableMessageTask.processMessage(AbstractCallableMessageTask.java:35)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:153)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:116)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
```

It should be possible to cleanly end such a query. Cleanly end might simply mean suppressing these exceptions.

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java
@@ -42,6 +42,7 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.CountDownLatch;
@@ -369,8 +370,14 @@ public class TaskletExecutionService {
                 }
                 progressTracker.mergeWith(result);
             } catch (Throwable e) {
-                logger.warning("Exception in " + t.tasklet, e);
-                t.executionTracker.exception(new JetException("Exception in " + t.tasklet + ": " + e, e));
+                if (e instanceof CancellationException) {
+                    logger.info("Job was cancelled by the user.");
+                    CancellationException ex = (CancellationException) e;
+                    t.executionTracker.exception(ex);
+                } else {
+                    logger.warning("Exception in " + t.tasklet, e);
+                    t.executionTracker.exception(new JetException("Exception in " + t.tasklet + ": " + e, e));
+                }
             } finally {
                 userMetricsContextContainer.setContext(null);
             }

--- a/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/RootResultConsumerSink.java
+++ b/hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/processors/RootResultConsumerSink.java
@@ -27,10 +27,14 @@ import com.hazelcast.jet.core.Watermark;
 import com.hazelcast.jet.sql.impl.JetQueryResultProducer;
 import com.hazelcast.jet.sql.impl.JetSqlCoreBackendImpl;
 import com.hazelcast.sql.impl.JetSqlCoreBackend;
+import com.hazelcast.sql.impl.QueryException;
 
 import javax.annotation.Nonnull;
 
+import java.util.concurrent.CancellationException;
+
 import static com.hazelcast.jet.core.ProcessorMetaSupplier.forceTotalParallelismOne;
+import static com.hazelcast.sql.impl.SqlErrorCode.CANCELLED_BY_USER;
 
 public final class RootResultConsumerSink implements Processor {
 
@@ -49,13 +53,27 @@ public final class RootResultConsumerSink implements Processor {
 
     @Override
     public boolean tryProcess() {
-        rootResultConsumer.ensureNotDone();
+        try {
+            rootResultConsumer.ensureNotDone();
+        } catch (QueryException e) {
+            if (e.getCode() == CANCELLED_BY_USER) {
+                throw new CancellationException();
+            }
+            throw e;
+        }
         return true;
     }
 
     @Override
     public void process(int ordinal, @Nonnull Inbox inbox) {
-        rootResultConsumer.consume(inbox);
+        try {
+            rootResultConsumer.consume(inbox);
+        } catch (QueryException e) {
+            if (e.getCode() == CANCELLED_BY_USER) {
+                throw new CancellationException();
+            }
+            throw e;
+        }
     }
 
     @Override

